component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,383 @@
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                StreamEvent clonedStreamEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                if (!streamEvent.getType().equals(ComplexEvent.Type.TIMER)) {
+                    Object groupEventMapKey = groupKeyExpressionExecutor.execute(clonedStreamEvent);
+                    ResourceStreamEventList resourceStreamEventList = groupEventMap.get(groupEventMapKey);
+                    if (resourceStreamEventList != null) {
+                        resourceStreamEventList.streamEventList.add(clonedStreamEvent);
+                    } else {
+                        List<StreamEvent> list = new ArrayList<>();
+                        list.add(clonedStreamEvent);
+                        groupEventMap.put(groupEventMapKey, new ResourceStreamEventList(list,
+                                clonedStreamEvent.getTimestamp() + timeInMilliSeconds));
+                        long newNextEmitTime = currentTime + timeInMilliSeconds;
+                        if (nextEmitTime == -1 || newNextEmitTime > nextEmitTime) {
+                            nextEmitTime = newNextEmitTime;
+                            if (scheduler != null) {
+                                scheduler.notifyAt(nextEmitTime);
+                            }
+                        }
+                    }
+                }
+                if (outputExpectsExpiredEvents) {
+                    for (Map.Entry<Object, ResourceStreamEventList> entry : groupEventMap.entrySet()) {
+                        windowLength = ResourceIdentifierStreamProcessor.getResourceCount(resourceName);
+                        if (entry.getValue().streamEventList.size() >= windowLength ||
+                                entry.getValue().expiryTimestamp < currentTime) {
+                            //update current event chunk with event batch
+                            for (StreamEvent event : entry.getValue().streamEventList) {
+                                currentEventChunk.add(event);
+                            }
+                            groupEventMap.remove(entry.getKey());
+                            //update outputStreamEventChunk with expired events chunk
+                            if (eventsToBeExpired.getFirst() != null) {
+                                while (eventsToBeExpired.hasNext()) {
+                                    StreamEvent expiredEvent = eventsToBeExpired.next();
+                                    expiredEvent.setTimestamp(currentTime);
+                                }
+                                outputStreamEventChunk.add(eventsToBeExpired.getFirst());
+                            }
+                            eventsToBeExpired.clear();
```

Comment:
remove this as "//update outputStreamEventChunk with expired events chunk" is already done

Reply:
Fixes added: d6242876caf0f65c7fd023c25c7c5e50e9fd7b3c